mllib/src/test/scala/org/apache/spark/mllib/linalg/MatricesSuite.scala

```diff
@@ -212,6 +284,17 @@ class MatricesSuite extends FunSuite {
     assert(deHorz1(2, 4) === 1.0)
     assert(deHorz1(1, 4) === 0.0)
 
+    // containing transposed matrices
+    val spHorzT = Matrices.horzcat(Array(spMat1TT, spMat2))
+    val spHorz2T = Matrices.horzcat(Array(spMat1TT, deMat2))
+    val spHorz3T = Matrices.horzcat(Array(deMat1TT, spMat2))
+    val deHorz1T = Matrices.horzcat(Array(deMat1TT, deMat2))
+
+    assert(deHorz1T.toBreeze === deHorz1.toBreeze)
```

Comment:
We shouldn't normally need toBreeze, but when I remove toBreeze, even though the matrices are exactly the same, I get an error. We shouldn't need to add `transpose` support to TestingUtils, because it already compares the output of `.toArray` of both matrices, which should be equal.